pkg/apis/core/validation/validation_test.go

```diff
@@ -10552,29 +10552,427 @@ func TestValidateServiceCreate(t *testing.T) {
 			},
 			numErrs: 1,
 		},
+		/* ip families validation */
+		{
+			name: "invalid, service with invalid ipFamilies",
+			tweakSvc: func(s *core.Service) {
+				invalidServiceIPFamily := core.IPFamily("not-a-valid-ip-family")
+				s.Spec.IPFamilies = []core.IPFamily{invalidServiceIPFamily}
+			},
+			numErrs: 1,
+		},
+		{
+			name: "invalid, service with invalid ipFamilies (2nd)",
+			tweakSvc: func(s *core.Service) {
+				invalidServiceIPFamily := core.IPFamily("not-a-valid-ip-family")
+				s.Spec.IPFamilyPolicy = &requireDualStack
+				s.Spec.IPFamilies = []core.IPFamily{core.IPv4Protocol, invalidServiceIPFamily}
+			},
+			numErrs:              1,
+			dualStackFeatureGate: true,
+		},
+		{
+			name: "IPFamilyPolicy:RequireDualStack is not set for two families",
```

Comment:
Defaulting will ensure this is not true - do we really want to validate it?  I am not sure if we do that for other things...

Reply:
yes - test case removed.